File: springmvc/spring-springmvc-hibernate/src/main/java/com/jiangKlijna/web/service/BaseService.java
package com.jiangKlijna.web.service;

import com.jiangKlijna.web.app.ContextWrapper;
import com.jiangKlijna.web.bean.Result;
import com.jiangKlijna.web.dao.IDao;

public class BaseService extends ContextWrapper {

	protected final IDao dao() {
		return getApplicationContext().getBean("dao", IDao.class);
	}

	protected Result sucessResult() {
		return getApplicationContext().getBean("sucessResult", Result.class);
	}

	protected Result sucessResult(Object data) {
		return sucessResult().setData(data);
	}

	protected Result errorResult(String message) {
		return getApplicationContext().getBean("errorResult", Result.class).setMessage(message);
	}
}

File: spring-springmvc-hibernate-kotlin/src/main/java/com/jiangKlijna/web/service/BaseService.kt
package com.jiangKlijna.web.service

import com.jiangKlijna.web.app.ContextWrapper
import com.jiangKlijna.web.bean.Result
import com.jiangKlijna.web.dao.IDao

open class BaseService : ContextWrapper() {

    protected fun dao(): IDao {
        return applicationContext.getBean("dao", IDao::class.java)
    }

    protected fun sucessResult(): Result {
        return applicationContext.getBean("sucessResult", Result::class.java)
    }

    protected fun sucessResult(data: Any): Result {
        return sucessResult().setData(data)
    }

    protected fun errorResult(message: String): Result {
        return applicationContext.getBean("errorResult", Result::class.java).setMessage(message)
    }
}
